Public/Src/Engine/Scheduler/PipHistoricPerfData.cs

```diff
@@ -176,23 +177,30 @@ public PipHistoricPerfData MakeFresh()
         /// </summary>
         public PipHistoricPerfData Merge(PipHistoricPerfData other)
         {
-            return new PipHistoricPerfData(
-                DefaultTimeToLive,
-                GetMergeResult(DurationInMs, other.DurationInMs),
-                GetMergeResult(PeakMemoryInKB, other.PeakMemoryInKB),
-                (ushort)GetMergeResult(ProcessorsInPercents, other.ProcessorsInPercents),
-                GetMergeResult(DiskIOInKB, other.DiskIOInKB));
+            var durationResult = GetMergeResult(DurationInMs, other.DurationInMs);
+            var peakMemoryResult = GetMergeResult(PeakMemoryInKB, other.PeakMemoryInKB);
+            var processorInPercentResult = GetMergeResult(ProcessorsInPercents, other.ProcessorsInPercents);
+            var diskIOResult = GetMergeResult(DiskIOInKB, other.DiskIOInKB);
+
+            return new PipHistoricPerfData(DefaultTimeToLive, durationResult, peakMemoryResult, (ushort)processorInPercentResult, diskIOResult);
         }
 
         private static uint GetMergeResult(uint oldData, uint newData)

```

Comment:
>GetMergeResult [](start = 28, length = 14)

How about making this internal and hitting it with some unit tests to bounds test the parameters?